src/dreyfus_httpd.erl

```diff
@@ -495,3 +510,64 @@ backoff_and_retry(Req, Db, DDoc, RetryCount, RetryPause) ->
             timer:sleep(RetryPause),
             handle_search_req(Req, Db, DDoc, RetryCount + 1, RetryPause * 2)
     end.
+
+
+partition_error(Msg) ->
+    throw({search_partition_error, Msg}).
+
+
+check_partition_restrictions(#index_query_args{} = Args) ->
+    #index_query_args{
+        partition_key= PartitionKey,
+        counts= Counts,
+        include_docs= IncludeDocs,
+        grouping = Grouping,
+        drilldown = Drilldown,
+        ranges = Ranges
+    } = Args,
+
+    #grouping{
+        by = GroupingBy
+    } = Grouping,
+
+    case PartitionKey of 
```

Comment:
I feel like there is a better way of doing all these checks. I also considered doing something like adding each field we checking into a list that contains the value it should equal and error message. Then looping over and checking the values, instead of all these case statements.

Reply:
you can probably create the list and run a ```lists:foreach``` on it to get the same desired effect